pkg/segment/results/blockresults/blockresult.go

```diff
@@ -842,14 +842,9 @@ func (gb *GroupByBuckets) ConvertToJson() (*GroupByBucketsJSON, error) {
 			Count:     bucket.count,
 			CurrStats: bucket.currStats,
 		}
-		retVals := make([]interface{}, 0, len(bucket.currStats))
-		for idx, rs := range bucket.runningStats {
-			if bucket.currStats[idx].MeasureFunc == utils.Cardinality {
-				encoded := rs.hll.ToBytes()
-				retVals = append(retVals, encoded)
-			} else {
-				retVals = append(retVals, rs.rawVal.CVal)
-			}
+		retVals := make([]RunningStatsJSON, 0, len(bucket.currStats))
+		for _, rs := range bucket.runningStats {
+			retVals = append(retVals, rs.GetRunningStatJSON())
 		}
```

Comment:
nit: you make the list have capacity `len(bucket.currStats)` but then append `len(bucket.runningStats)` items to it. Why not init it with capacity `len(bucket.runningStats)`?

Same comment for line 866

Reply:
I just followed what was before, we could change it. What is your opinion?

Reply:
addressed in the latest commit.